src/dao/user.dao.ts

```diff
@@ -28,6 +28,23 @@ class UserDAO {
     });
   }
 
+  async fetchUsers(
+    filter: { username?: string; email?: string }, 
+    page: number = 0, 
+    pageSize: number = 10
+  ) {
+    const skip = (page - 1) * pageSize;
```

Comment:
What happends if `page = 0`?

Reply:
if `page === 0` , `skip` will be a negative number `-1 * pageSize`.

We should consider handling this scenario